### DISCUSSION TITLE
Does Dex have support for CBL Mariner aka Mariner aks "Azure Linux"?

### DISCUSSION THREAD

**Discussion by pfarikrispy** [dexidp/dex#3284] [Q&A] [answered]
We recently tried to deploy a test instance of Dex on our dev AKS cluster and ran into the strangest behavior :) 

* Although we deployed the latest helm chart, the diff showed that the intention was to install the 0.1.0 version. That's earliest, ancient version of dex.
* When we tried the same on another cluster, the intention was correct and chart 0.15.3 was chosen.

The only difference between the cluster was that our own dev cluster is using the new container-optimized "Azure Linux" for its default node pool where the second cluster is still using "Ubuntu Linux" for those nodes.

This leads me to ask: does Dex support running on/with Mariner aka "[Azure Linux](https://learn.microsoft.com/en-us/azure/azure-linux/intro-azure-linux)"? Have you tried? Do you know about Mariner? Or we wading into a deep unknown territory here? :)

**Comment by pfarikrispy** [comment, marked as answer]
I think we solved this. We cleared some local caches and then it seemed to work as expected.

**Reply by nabokihms** [reply]
Good to hear. 

@pfarikrispy as for me, I've never heard about Azure Linux, but it does not seem that Dex has Linux distribution-specific dependencies other than any go application has.

**Comment by pfarikrispy** [comment]
Thanks Maksim, I didn't think so either but we had such weird errors and
only the OS used in the node pool was a real difference.
Turned out to be cache dumminess :D

Urge you to try the newer container-optimized OSes in your k8s clusters, as
they remove a ton of vulnerabilities. They only provide a container-runtime
and do away with the traditional OS layer. "Azure Linux" for AKS,
BottleRocket for EKS, not sure about GKE since we don't use that (yet). You
can still provide Ubuntu in a node group, when you need to. Very cool and -
as you said - most things "just work". :)
Cheers,
Christian.
